### PR TITLE
[vpj] Support AAWC TTL repush with compression

### PR DESCRIPTION
**Pull request by sixpluszero**
## [vpj] Support AAWC TTL repush with compression
Current AAWC TTL repush need to deserialize value and perform a TTL based delete on the record to see if it needs to be filtered. The code does not handle the case when source version has compression enabled thus deserialization can fail. This PR fixes it by building the source version compressor and fix the deserialization issue.

## How was this PR tested?
Add Compression Mode for AAWC TTL integration test.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by majisourav99 — COMMENTED** [review]
Change looks good to me, can you add some tests with ZSTD compression ?

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java [hunk 1]
```diff
@@ -125,7 +152,11 @@ boolean filterByTTLandMaybeUpdateValue(final INPUT_VALUE value) {
         valueSerializerCache.computeIfAbsent(valueSchemaId, this::generateValueSerializer);
     RecordSerializer<GenericRecord> rmdSerializer =
         rmdSerializerCache.computeIfAbsent(rmdVersionId, this::generateRmdSerializer);
-    updateValuePayload(value, valueSerializer.serialize(valueRecord));
+    try {
+      updateValuePayload(value, sourceVersionCompressor.compress(valueSerializer.serialize(valueRecord)));
```
Comment (gaojieliu) [inline clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java:156]:
```
props.getBoolean(KAFKA_INPUT_COMPRESSION_BUILD_NEW_DICT_ENABLED, true);
```
Repush will always build a new dict for the repushed version, so shall we read the dictionary in the new version instead for the compression?

Comment (sixpluszero) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java:156]: Hmm, would that be available in the mapper / reducer phase? As far as I know this filter is invoked in both mapper and reducer today. 

Also, what'd be the difference for reading for the new dictionary? I am a bit new to ZSTD_WITH_DICT so I don't know that difference.
Also I am wondering, if dest version changed compression strategy, would it still work?

Comment (majisourav99) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java:156]: Using newer dictionary would give better compression, but not a must. If dest changed compression strategy, we should fail the repush?

Comment (gaojieliu) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java:156]: I think we can build a new compressor for the dest version as the regular push can get a proper compressor in the reducer phase.
You can check the related code.

**Review by gaojieliu — APPROVED** [review]
Synced offline, and we can optimize the multiple rounds of decompression/re-compression in a separate PR.

**Comment by sixpluszero** [comment]
> Change looks good to me, can you add some tests with ZSTD compression ?

The newly added integration test does cover all compression

**Pull request merged**